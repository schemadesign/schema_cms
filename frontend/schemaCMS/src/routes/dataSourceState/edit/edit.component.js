import React, { Fragment, PureComponent } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import Helmet from 'react-helmet';
import { Stepper } from 'schemaUI';

import { LinkContainer } from './edit.styles';
import messages from './edit.messages';
import { contentStyles, NavigationButtons } from '../../../shared/components/navigationStyles';
import { filterMenuOptions } from '../../../shared/utils/helpers';
import reportError from '../../../shared/utils/reportError';
import { getProjectMenuOptions, PROJECT_STATE_ID } from '../../project/project.constants';
import { MobileMenu } from '../../../shared/components/menu/mobileMenu';
import { ContextHeader } from '../../../shared/components/contextHeader';
import { DataSourceStateForm } from '../../../shared/components/dataSourceStateForm';
import { BackButton, NavigationContainer, NextButton } from '../../../shared/components/navigation';
import { Modal, ModalActions, modalStyles, ModalTitle } from '../../../shared/components/modal/modal.styles';
import { Link } from '../../../theme/typography';

export class Edit extends PureComponent {
  static propTypes = {
    handleSubmit: PropTypes.func.isRequired,
    removeState: PropTypes.func.isRequired,
    isSubmitting: PropTypes.bool.isRequired,
    dirty: PropTypes.bool.isRequired,
    userRole: PropTypes.string.isRequired,
    state: PropTypes.object.isRequired,
    intl: PropTypes.object.isRequired,
    history: PropTypes.shape({
      push: PropTypes.func.isRequired,
    }),
  };

  state = {
    confirmationModalOpen: false,
    removeLoading: false,
  };

  handleCancel = () => this.props.history.push(`/project/${this.props.state.project}/state`);

  handleSubmit = () => {
    const { dirty, handleSubmit, history, state } = this.props;
    const redirectUrl = `/state/${state.id}/tags`;

    if (dirty) {
      return handleSubmit();
    }

    return history.push(redirectUrl);
  };

  handleRemoveState = () => this.setState({ confirmationModalOpen: true });

  handleCancelRemove = () => this.setState({ confirmationModalOpen: false });

  handleConfirmRemove = async () => {
    try {
      this.setState({ removeLoading: true });

      const { id, project } = this.props.state;
      await this.props.removeState({ stateId: id, projectId: project });
    } catch (error) {
      this.setState({ removeLoading: false });
      reportError(error);
    }
  };

  render() {
    const { userRole, isSubmitting, state } = this.props;
    const { confirmationModalOpen, removeLoading } = this.state;
    const projectId = state.project;
    const menuOptions = getProjectMenuOptions(projectId);
    const title = state.name;

    return (
      <Fragment>
        <Helmet title={title} />
        <MobileMenu
          headerTitle={title}
          headerSubtitle={<FormattedMessage {...messages.subTitle} />}
          options={filterMenuOptions(menuOptions, userRole)}
          active={PROJECT_STATE_ID}
        />
        <ContextHeader title={title} subtitle={<FormattedMessage {...messages.subTitle} />} />
        <DataSourceStateForm {...this.props} />
        <LinkContainer>
          <Link onClick={this.handleRemoveState}>
            <FormattedMessage {...messages.deleteState} />
          </Link>
        </LinkContainer>
        <NavigationContainer fixed contentStyles={contentStyles}>
          <NavigationButtons>
            <BackButton type="button" onClick={this.handleCancel}>
              <FormattedMessage {...messages.cancel} />
            </BackButton>
            <NextButton type="submit" onClick={this.handleSubmit} loading={isSubmitting} disabled={isSubmitting} />
          </NavigationButtons>
          <Stepper steps={3} activeStep={1} />
        </NavigationContainer>
        <Modal isOpen={confirmationModalOpen} contentLabel="Confirm Removal" style={modalStyles}>
          <ModalTitle>
            <FormattedMessage {...messages.removeTitle} />
          </ModalTitle>
          <ModalActions>
            <BackButton onClick={this.handleCancelRemove} disabled={removeLoading}>
              <FormattedMessage {...messages.cancelRemoval} />
            </BackButton>
            <NextButton
              id="confirmRemovalBtn"
              onClick={this.handleConfirmRemove}
              loading={removeLoading}
              disabled={removeLoading}
            >
              <FormattedMessage {...messages.confirmRemoval} />
            </NextButton>
          </ModalActions>
        </Modal>
      </Fragment>
    );
  }
}
