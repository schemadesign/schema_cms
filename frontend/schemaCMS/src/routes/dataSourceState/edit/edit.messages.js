/* eslint-disable max-len */
import { defineMessages } from 'react-intl';

export default defineMessages({
  subTitle: {
    id: 'dataSourceState.edit.subTitle',
    defaultMessage: 'Edit State',
  },
  cancel: {
    id: 'dataSourceState.edit.cancel',
    defaultMessage: 'Cancel',
  },
  deleteState: {
    id: 'dataSourceState.edit.deleteState',
    defaultMessage: 'Remove State',
  },
  confirmRemoval: {
    id: 'dataSourceState.edit.confirmRemoval',
    defaultMessage: 'Confirm',
  },
  removeTitle: {
    id: 'dataSourceState.edit.removeTitle',
    defaultMessage: 'Are you sure you want to remove this state?',
  },
  cancelRemoval: {
    id: 'dataSourceState.edit.cancelRemoval',
    defaultMessage: 'Cancel',
  },
  nameStateNameNotUniqueError: {
    id: 'dataSourceState.edit.nameStateNameNotUniqueError',
    defaultMessage: 'A state with this name already exists in project.',
  },
});
