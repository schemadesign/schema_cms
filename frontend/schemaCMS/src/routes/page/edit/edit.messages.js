/* eslint-disable max-len */
import { defineMessages } from 'react-intl';

export default defineMessages({
  title: {
    id: 'page.edit.title',
    defaultMessage: 'Page',
  },
  subTitle: {
    id: 'page.edit.subTitle',
    defaultMessage: 'Edit',
  },
  save: {
    id: 'page.edit.save',
    defaultMessage: 'Save',
  },
  back: {
    id: 'page.edit.back',
    defaultMessage: 'Back',
  },
  titlePageNameUniqueError: {
    id: 'page.edit.titlePageNameUniqueError',
    defaultMessage: 'Page with this name already exist in folder.',
  },
  removePage: {
    id: 'page.edit.removePage',
    defaultMessage: 'Remove Page',
  },
  cancelRemoval: {
    id: 'page.edit.cancelRemoval',
    defaultMessage: 'Cancel',
  },
  confirmRemoval: {
    id: 'page.edit.confirmRemoval',
    defaultMessage: 'Confirm',
  },
  removeTitle: {
    id: 'page.edit.removeTitle',
    defaultMessage: 'Removing a page will also remove any content under it. Are you sure you want to proceed?',
  },
});
