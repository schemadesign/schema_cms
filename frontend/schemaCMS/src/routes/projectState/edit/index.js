export { default as Edit } from './edit.container';
