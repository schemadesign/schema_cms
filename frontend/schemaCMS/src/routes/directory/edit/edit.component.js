import React, { Fragment, PureComponent } from 'react';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';
import { always, path } from 'ramda';

import { Form } from './edit.styles';
import messages from './edit.messages';
import { TextInput } from '../../../shared/components/form/inputs/textInput';
import { DIRECTORY_NAME } from '../../../modules/directory/directory.constants';
import { TopHeader } from '../../../shared/components/topHeader';
import { ContextHeader } from '../../../shared/components/contextHeader';
import { BackButton, NavigationContainer, NextButton } from '../../../shared/components/navigation';
import { renderWhenTrueOtherwise } from '../../../shared/utils/rendering';
import { Loader } from '../../../shared/components/loader';

export class Edit extends PureComponent {
  static propTypes = {
    intl: PropTypes.object.isRequired,
    values: PropTypes.object.isRequired,
    directory: PropTypes.object.isRequired,
    isValid: PropTypes.bool.isRequired,
    handleSubmit: PropTypes.func.isRequired,
    handleChange: PropTypes.func.isRequired,
    handleBlur: PropTypes.func.isRequired,
    fetchDirectory: PropTypes.func.isRequired,
    match: PropTypes.shape({
      params: PropTypes.shape({
        directoryId: PropTypes.string.isRequired,
      }),
    }),
    history: PropTypes.shape({
      push: PropTypes.func.isRequired,
    }),
  };

  state = {
    loading: true,
  };

  async componentDidMount() {
    try {
      const directoryId = path(['match', 'params', 'directoryId'], this.props);
      await this.props.fetchDirectory({ directoryId });
      this.setState({ loading: false });
    } catch (e) {
      this.props.history.push('/');
    }
  }

  handleBackClick = () => this.props.history.push(`/project/${path(['directory', 'project'], this.props)}/directory`);

  renderContent = renderWhenTrueOtherwise(always(<Loader />), () => (
    <TextInput
      value={this.props.values[DIRECTORY_NAME]}
      onChange={this.props.handleChange}
      name={DIRECTORY_NAME}
      label={this.props.intl.formatMessage(messages.directoryFieldName)}
      placeholder={this.props.intl.formatMessage(messages.directoryFieldPlaceholder)}
      fullWidth
      isEdit
      {...this.props}
    />
  ));

  render() {
    const { handleSubmit, isValid } = this.props;
    const { loading } = this.state;
    const headerTitle = <FormattedMessage {...messages.title} />;
    const headerSubtitle = <FormattedMessage {...messages.subTitle} />;

    return (
      <Fragment>
        <TopHeader headerTitle={headerTitle} headerSubtitle={headerSubtitle} />
        <ContextHeader title={headerTitle} subtitle={headerSubtitle} />
        <Form onSubmit={handleSubmit}>
          {this.renderContent(loading)}
          <NavigationContainer>
            <BackButton id="backBtn" onClick={this.handleBackClick}>
              <FormattedMessage {...messages.cancel} />
            </BackButton>
            <NextButton id="editDirectoryBtn" type="submit" disabled={!isValid}>
              <FormattedMessage {...messages.createDirectory} />
            </NextButton>
          </NavigationContainer>
        </Form>
      </Fragment>
    );
  }
}
