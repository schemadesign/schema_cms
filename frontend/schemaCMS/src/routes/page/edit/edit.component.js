import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import { Container } from './edit.styles';

export class Edit extends PureComponent {
  static propTypes = {
    page: PropTypes.object.isRequired,
  };

  render() {
    return <Container>{this.props.page.id}</Container>;
  }
}
