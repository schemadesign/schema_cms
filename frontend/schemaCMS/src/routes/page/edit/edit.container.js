import { connect } from 'react-redux';
import { promisifyRoutine, bindPromiseCreators } from 'redux-saga-routines';
import { createStructuredSelector } from 'reselect';
import { withRouter } from 'react-router-dom';
import { hot } from 'react-hot-loader';
import { compose, path } from 'ramda';
import { injectIntl } from 'react-intl';
import { withFormik } from 'formik';

import { Edit } from './edit.component';
import { PageRoutines, selectPage } from '../../../modules/page';
import { PAGE_FORM, INITIAL_VALUES, PAGE_SCHEMA } from '../../../modules/page/page.constants';
import { errorMessageParser } from '../../../shared/utils/helpers';

const mapStateToProps = createStructuredSelector({
  page: selectPage,
});

export const mapDispatchToProps = dispatch =>
  bindPromiseCreators(
    {
      updatePage: promisifyRoutine(PageRoutines.update),
    },
    dispatch
  );

export default compose(
  hot(module),
  connect(
    mapStateToProps,
    mapDispatchToProps
  ),
  withRouter,
  injectIntl,
  withFormik({
    displayName: PAGE_FORM,
    enableReinitialize: true,
    mapPropsToValues: ({ page }) => ({ ...INITIAL_VALUES, ...page }),
    validationSchema: () => PAGE_SCHEMA,
    handleSubmit: async (data, { props, setSubmitting, setErrors }) => {
      try {
        setSubmitting(true);
        const pageId = path(['match', 'params', 'pageId'], props);
        const directoryId = path(['page', 'directory', 'id'], props);

        await props.updatePage({ pageId, directoryId, ...data });
      } catch (errors) {
        const { formatMessage } = props.intl;
        const errorMessages = errorMessageParser({ errors, messages, formatMessage });

        setErrors(errorMessages);
      } finally {
        setSubmitting(false);
      }
    },
  })
)(Edit);
