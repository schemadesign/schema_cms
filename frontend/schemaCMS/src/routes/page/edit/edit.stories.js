import React from 'react';
import { storiesOf } from '@storybook/react';

import { Edit } from './edit.component';
import { withTheme } from '../../../.storybook/decorators';
import { history, intl } from '../../../.storybook/helpers';
import { ROLES } from '../../../modules/userProfile/userProfile.constants';

export const defaultProps = {
  userRole: ROLES.ADMIN,
  page: { folder: { id: '1' } },
  values: {
    title: 'title',
    description: 'description',
    keywords: 'keywords',
  },
  fetchPage: Function.prototype,
  handleChange: Function.prototype,
  handleSubmit: Function.prototype,
  removePage: Function.prototype,
  isValid: true,
  isSubmitting: false,
  history,
  intl,
  match: {
    params: {
      pageId: '1',
    },
  },
};

storiesOf('Page|Edit', module)
  .addDecorator(withTheme())
  .add('Default', () => <Edit {...defaultProps} />);
