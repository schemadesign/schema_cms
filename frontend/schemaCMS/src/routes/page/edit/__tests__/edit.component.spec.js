import React from 'react';
import { shallow } from 'enzyme';

import { Edit } from '../edit.component';
import { defaultProps } from '../edit.stories';
import { Form } from '../edit.styles';

describe('Edit: Component', () => {
  const component = props => <Edit {...defaultProps} {...props} />;

  const render = (props = {}) => shallow(component(props));

  it('should render correctly with loader', () => {
    const wrapper = render();
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly', async () => {
    defaultProps.fetchPage = jest.fn().mockReturnValue(Promise.resolve());
    const wrapper = render();
    await Promise.resolve();
    global.expect(wrapper).toMatchSnapshot();
  });

  it('should return to folder list', () => {
    jest.spyOn(defaultProps.history, 'push');
    const wrapper = render();
    wrapper.find('#backBtn').simulate('click');

    expect(defaultProps.history.push).toHaveBeenCalledWith('/folder/1');
  });

  it('should submit form', () => {
    jest.spyOn(defaultProps, 'handleSubmit');
    const wrapper = render();
    wrapper.find(Form).simulate('submit');

    expect(defaultProps.handleSubmit).toHaveBeenCalled();
  });

  it('should call removePage on confirm button click', () => {
    jest.spyOn(defaultProps, 'removePage');
    const wrapper = render();
    wrapper.find('#confirmRemovalBtn').simulate('click');

    expect(defaultProps.removePage).toHaveBeenCalledWith({ pageId: '1', folderId: '1' });
  });
});
