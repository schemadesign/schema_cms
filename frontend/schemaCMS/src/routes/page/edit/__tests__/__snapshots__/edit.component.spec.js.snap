// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit: Component should render correctly 1`] = `
<Fragment>
  <TopHeader
    headerSubtitle={
      <FormattedMessage
        defaultMessage="Edit"
        id="page.edit.subTitle"
        values={Object {}}
      />
    }
    headerTitle={
      <FormattedMessage
        defaultMessage="Page"
        id="page.edit.title"
        values={Object {}}
      />
    }
  />
  <ContextHeader
    subtitle={
      <FormattedMessage
        defaultMessage="Edit"
        id="page.edit.subTitle"
        values={Object {}}
      />
    }
    title={
      <FormattedMessage
        defaultMessage="Page"
        id="page.edit.title"
        values={Object {}}
      />
    }
  />
  <LoadingWrapper
    error={null}
    loading={false}
    noData={false}
    noDataContent={null}
  >
    <styled.form
      onSubmit={[Function]}
    >
      <PageForm
        fetchPage={
          [MockFunction] {
            "calls": Array [
              Array [
                Object {
                  "pageId": "1",
                },
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": Promise {},
              },
            ],
          }
        }
        handleChange={[Function]}
        handleSubmit={[Function]}
        history={
          Object {
            "push": [Function],
          }
        }
        intl={
          Object {
            "formatMessage": [Function],
          }
        }
        isSubmitting={false}
        isValid={true}
        match={
          Object {
            "params": Object {
              "pageId": "1",
            },
          }
        }
        page={
          Object {
            "folder": Object {
              "id": "1",
            },
          }
        }
        removePage={[Function]}
        values={
          Object {
            "description": "description",
            "keywords": "keywords",
            "title": "title",
          }
        }
      />
      <Styled(Span)
        id="removePageDesktopBtn"
        onClick={[Function]}
      >
        <FormattedMessage
          defaultMessage="Remove Page"
          id="page.edit.removePage"
          values={Object {}}
        />
      </Styled(Span)>
      <NavigationContainer
        hideOnDesktop={false}
        right={false}
      >
        <BackButton
          disabled={false}
          id="backBtn"
          onClick={[Function]}
        >
          <FormattedMessage
            defaultMessage="Back"
            id="page.edit.back"
            values={Object {}}
          />
        </BackButton>
        <NextButton
          disabled={false}
          id="savePageBtn"
          loading={false}
          type="submit"
        >
          <FormattedMessage
            defaultMessage="Save"
            id="page.edit.save"
            values={Object {}}
          />
        </NextButton>
      </NavigationContainer>
    </styled.form>
  </LoadingWrapper>
  <Styled(Modal)
    contentLabel="Confirm Removal"
    style={
      Object {
        "content": Object {
          "WebkitOverflowScrolling": "touch",
          "alignItems": "center",
          "border": null,
          "borderRadius": "4px",
          "boxShadow": "0 4px 8px 0 rgba(0, 0, 0, 0.5)",
          "display": "flex",
          "flexDirection": "column",
          "left": "50%",
          "maxWidth": "600px",
          "outline": "none",
          "overflow": "auto",
          "position": "absolute",
          "right": "10px",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "width": "calc(100% - 20px)",
        },
        "overlay": Object {
          "backgroundColor": "rgba(29, 29, 32, 0.75)",
          "bottom": 0,
          "left": 0,
          "position": "fixed",
          "right": 0,
          "top": 0,
          "zIndex": 10000,
        },
      }
    }
  >
    <Styled(H1)>
      <FormattedMessage
        defaultMessage="Removing a page will also remove any content under it. Are you sure you want to proceed?"
        id="page.edit.removeTitle"
        values={Object {}}
      />
    </Styled(H1)>
    <styled.div>
      <BackButton
        disabled={false}
        onClick={[Function]}
      >
        <FormattedMessage
          defaultMessage="Cancel"
          id="page.edit.cancelRemoval"
          values={Object {}}
        />
      </BackButton>
      <NextButton
        id="confirmRemovalBtn"
        loading={false}
        onClick={[Function]}
      >
        <FormattedMessage
          defaultMessage="Confirm"
          id="page.edit.confirmRemoval"
          values={Object {}}
        />
      </NextButton>
    </styled.div>
  </Styled(Modal)>
</Fragment>
`;

exports[`Edit: Component should render correctly with loader 1`] = `
<Fragment>
  <TopHeader
    headerSubtitle={
      <FormattedMessage
        defaultMessage="Edit"
        id="page.edit.subTitle"
        values={Object {}}
      />
    }
    headerTitle={
      <FormattedMessage
        defaultMessage="Page"
        id="page.edit.title"
        values={Object {}}
      />
    }
  />
  <ContextHeader
    subtitle={
      <FormattedMessage
        defaultMessage="Edit"
        id="page.edit.subTitle"
        values={Object {}}
      />
    }
    title={
      <FormattedMessage
        defaultMessage="Page"
        id="page.edit.title"
        values={Object {}}
      />
    }
  />
  <LoadingWrapper
    error={null}
    loading={true}
    noData={false}
    noDataContent={null}
  >
    <styled.form
      onSubmit={[Function]}
    >
      <PageForm
        fetchPage={[Function]}
        handleChange={[Function]}
        handleSubmit={[Function]}
        history={
          Object {
            "push": [Function],
          }
        }
        intl={
          Object {
            "formatMessage": [Function],
          }
        }
        isSubmitting={false}
        isValid={true}
        match={
          Object {
            "params": Object {
              "pageId": "1",
            },
          }
        }
        page={
          Object {
            "folder": Object {
              "id": "1",
            },
          }
        }
        removePage={[Function]}
        values={
          Object {
            "description": "description",
            "keywords": "keywords",
            "title": "title",
          }
        }
      />
      <Styled(Span)
        id="removePageDesktopBtn"
        onClick={[Function]}
      >
        <FormattedMessage
          defaultMessage="Remove Page"
          id="page.edit.removePage"
          values={Object {}}
        />
      </Styled(Span)>
      <NavigationContainer
        hideOnDesktop={false}
        right={false}
      >
        <BackButton
          disabled={false}
          id="backBtn"
          onClick={[Function]}
        >
          <FormattedMessage
            defaultMessage="Back"
            id="page.edit.back"
            values={Object {}}
          />
        </BackButton>
        <NextButton
          disabled={false}
          id="savePageBtn"
          loading={false}
          type="submit"
        >
          <FormattedMessage
            defaultMessage="Save"
            id="page.edit.save"
            values={Object {}}
          />
        </NextButton>
      </NavigationContainer>
    </styled.form>
  </LoadingWrapper>
  <Styled(Modal)
    contentLabel="Confirm Removal"
    style={
      Object {
        "content": Object {
          "WebkitOverflowScrolling": "touch",
          "alignItems": "center",
          "border": null,
          "borderRadius": "4px",
          "boxShadow": "0 4px 8px 0 rgba(0, 0, 0, 0.5)",
          "display": "flex",
          "flexDirection": "column",
          "left": "50%",
          "maxWidth": "600px",
          "outline": "none",
          "overflow": "auto",
          "position": "absolute",
          "right": "10px",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "width": "calc(100% - 20px)",
        },
        "overlay": Object {
          "backgroundColor": "rgba(29, 29, 32, 0.75)",
          "bottom": 0,
          "left": 0,
          "position": "fixed",
          "right": 0,
          "top": 0,
          "zIndex": 10000,
        },
      }
    }
  >
    <Styled(H1)>
      <FormattedMessage
        defaultMessage="Removing a page will also remove any content under it. Are you sure you want to proceed?"
        id="page.edit.removeTitle"
        values={Object {}}
      />
    </Styled(H1)>
    <styled.div>
      <BackButton
        disabled={false}
        onClick={[Function]}
      >
        <FormattedMessage
          defaultMessage="Cancel"
          id="page.edit.cancelRemoval"
          values={Object {}}
        />
      </BackButton>
      <NextButton
        id="confirmRemovalBtn"
        loading={false}
        onClick={[Function]}
      >
        <FormattedMessage
          defaultMessage="Confirm"
          id="page.edit.confirmRemoval"
          values={Object {}}
        />
      </NextButton>
    </styled.div>
  </Styled(Modal)>
</Fragment>
`;
